crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -1655,6 +1653,61 @@ WHERE e1.epoch = e2.epoch
 
         epoch_info.to_epoch_info(validators)
     }
+
+    /// address stats methods
+    async fn get_last_address_processed_checkpoint(&self) -> Result<i64, IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            address_stats::dsl::address_stats
+                .select(max(address_stats::checkpoint))
+                .first::<Option<i64>>(conn)
+                // -1 to differentiate between no checkpoints and the first checkpoint
+                .map(|o| o.unwrap_or(-1))
+        })
+        .context("Failed reading latest object checkpoint from PostgresDB")
+    }
+
+    async fn calculate_address_stats(&self, checkpoint: i64) -> Result<AddressStats, IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            let cp: Checkpoint = checkpoints::dsl::checkpoints
+                .filter(checkpoints::sequence_number.eq(checkpoint))
+                .limit(1)
+                .first(conn)?;
+            let cp_timestamp_ms = cp.timestamp_ms;
+            let cumulative_addresses = addresses::dsl::addresses
+                .filter(addresses::first_appearance_time.le(cp_timestamp_ms))
+                .select(count(addresses::account_address))
+                .first(conn)?;
+            let cumulative_active_addresses = active_addresses::dsl::active_addresses
+                .filter(active_addresses::first_appearance_time.le(cp_timestamp_ms))
+                .select(count(active_addresses::account_address))
+                .first(conn)?;
+            let time_one_day_ago = cp_timestamp_ms - 1000 * 60 * 60 * 24;
+            let daily_active_addresses = active_addresses::dsl::active_addresses
+                .filter(active_addresses::first_appearance_time.le(cp_timestamp_ms))
```

Comment:
this would be a rolling daily avg? cool

Reply:
it's a rolling one indeed, we also roughly have time-series data for address count.